DeepSpeech.py

```diff
@@ -212,20 +202,25 @@
 # ==============
 
 def variable_on_cpu(name, shape, initializer):
-    r"""
+    r'''
     Next we concern ourselves with graph creation.
     However, before we do so we must introduce a utility function ``variable_on_cpu()``
     used to create a variable in CPU memory.
-    """
-    # Use the /cpu:0 device for scoped operations
-    with tf.device('/cpu:0'):
+    '''
+    # Use the /cpu:0 device on worker_device for scoped operations
+    if len(FLAGS.ps_hosts) == 0:
+        device = worker_device
```

Comment:
Shouldn't this be `cpu_device`? Also, these globals being used throughout the code but only defined in `main` are kind of hard to read.